src/components/QuranReader/ReadingView/Page.tsx

```diff
@@ -16,9 +16,14 @@ type PageProps = {
 
 const Page = ({ verses, page }: PageProps) => {
   const lines = useMemo(() => groupLinesByVerses(verses), [verses]);
-  const { quranTextFontScale } = useSelector(selectQuranReaderStyles) as QuranReaderStyles;
-  const { showWordByWordTranslation, showWordByWordTransliteration } =
-    useSelector(selectReadingPreferences);
+  const { quranTextFontScale } = useSelector(
+    selectQuranReaderStyles,
+    shallowEqual,
+  ) as QuranReaderStyles;
+  const { showWordByWordTranslation, showWordByWordTransliteration } = useSelector(
```

Comment:
In this case, does it only compare `showWordByWordTranslation` and `showWordByWordTransliteration` or the entire object returned from `selectReadingPreferences`? because if it is comparing the entire object, I think shallowEqual will not be useful as the object contains an array (`showTooltipFor`).